webpack/InsightsCloudSync/Components/InsightsTable/InsightsTableActions.js

```diff
@@ -51,15 +54,23 @@ export const fetchInsights = (queryParams = {}) => (dispatch, getState) => {
     })
   );
 
-  const isAllSelected = selectIsAllSelected(state);
-  if (isAllSelected) {
+  if (!isSelectAll) {
     dispatch(clearAllSelection());
-  } else {
-    dispatch(setSelectAllAlert(false));
-    dispatch(setSelectAll(false));
   }
 };
 
+const selectAllIdsFromResonse = (dispatch, response) => {
+  const selectedIds = {};
+  const results = response?.data?.hits || [];
```

Comment:
no need, because on success we will get the response.data object in any case
```suggestion
  const results = response.data.hits || [];
```

Reply:
Sure I'll change to that, do you have any resource where I can read about that? :)

Reply:
that's what Axios and the API middleware will return on success,
what's in the data object is what returned from the server in the render JSON,
and in this case, if hits are undefined it will return an empty array